CMakeLists.txt

```diff
@@ -0,0 +1,658 @@
+# Copyright (c) 2023-present The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or https://opensource.org/license/mit/.
+
+# Ubuntu 22.04 LTS Jammy Jellyfish, https://wiki.ubuntu.com/Releases, EOSS in June 2027:
+#  - CMake 3.22.1, https://packages.ubuntu.com/jammy/cmake
+#
+# Centos Stream 9, EOL in May 2027:
+#  - CMake 3.26.5, https://mirror.stream.centos.org/9-stream/AppStream/x86_64/os/Packages/
+cmake_minimum_required(VERSION 3.22)
+if(POLICY CMP0141)
+  # MSVC debug information format flags are selected by an abstraction.
+  # We want to use the CMAKE_MSVC_DEBUG_INFORMATION_FORMAT variable
+  # to select the MSVC debug information format.
+  cmake_policy(SET CMP0141 NEW)
+endif()
+
+#=============================
+# Project / Package metadata
+#=============================
+set(PACKAGE_NAME "Bitcoin Core")
+set(CLIENT_VERSION_MAJOR 27)
+set(CLIENT_VERSION_MINOR 99)
+set(CLIENT_VERSION_BUILD 0)
+set(CLIENT_VERSION_RC 0)
+set(CLIENT_VERSION_IS_RELEASE "false")
+set(COPYRIGHT_YEAR "2024")
+
+# During the enabling of the CXX and CXXOBJ languages, we modify
+# CMake's compiler/linker invocation strings by appending the content
+# of the user-defined `APPEND_*` variables, which allows overriding
+# any flag. We also ensure that the APPEND_* flags are considered
+# during CMake's tests, which use the `try_compile()` command.
+#
+# CMake's docs state that the `CMAKE_TRY_COMPILE_PLATFORM_VARIABLES`
+# variable "is meant to be set by CMake's platform information modules
+# for the current toolchain, or by a toolchain file." We do our best
+# to set it before the `project()` command.
+set(CMAKE_TRY_COMPILE_PLATFORM_VARIABLES
+  CMAKE_CXX_COMPILE_OBJECT
+  CMAKE_OBJCXX_COMPILE_OBJECT
+  CMAKE_CXX_LINK_EXECUTABLE
+)
+
+project(BitcoinCore
```

Comment:
Could you elaborate please? What are the benefits of maintaining the build system in separate projects? Is this practice used by other open-source projects?

Reply:
Might be controversial, but other Bitcoin clones seem to split into multiple projects after their CMake migrations: 
* https://github.com/Bitcoin-ABC/bitcoin-abc/blob/master/src/bench/CMakeLists.txt#L3
* https://github.com/bitcoin-cash-node/bitcoin-cash-node/blob/master/src/bench/CMakeLists.txt#L3
* https://github.com/bitcoin-sv/bitcoin-sv/blob/master/src/test/CMakeLists.txt#L5

Reply:
Still curios about:
> What are the benefits of maintaining the build system in separate projects?

Reply:
I'm also curious, that's what I'm asking basically, why did the other clones decide to do this differently.

I could only find general `Modularity and Reusability/Scalability/Improved Dependency Management/Isolation of Build Configurations` arguments online, but not sure any applies here, hence my question.

Reply:
> Might be controversial, but other Bitcoin clones seem to split into multiple projects after their CMake migrations:
> 
>    * https://github.com/Bitcoin-ABC/bitcoin-abc/blob/master/src/bench/CMakeLists.txt#L3

It is not controversial, just broken :)

The `cmake -S src/bench -B build` command fails.

Reply:
> The cmake -S src/bench -B build command fails.

It seems to me [they're using `ninja`](https://github.com/bitcoin-cash-node/bitcoin-cash-node/blob/master/doc/benchmarking.md?plain=1#L8) for these tasks, i.e. [`ninja bench-bitcoin`](https://github.com/bitcoin-cash-node/bitcoin-cash-node/blob/master/doc/ninja_targets.md?plain=1#L251) instead of raw cmake.

Reply:
> > The cmake -S src/bench -B build command fails.
> 
> It seems to me [they're using `ninja`](https://github.com/bitcoin-cash-node/bitcoin-cash-node/blob/master/doc/benchmarking.md?plain=1#L8) for these tasks, i.e. [`ninja bench-bitcoin`](https://github.com/bitcoin-cash-node/bitcoin-cash-node/blob/master/doc/ninja_targets.md?plain=1#L251) instead of raw cmake.

It seems this discussion has gone off topic :)

Reply:
Didn't realize that, please resolve it in that case